### DISCUSSION TITLE
Experimental replication parameters

### DISCUSSION THREAD

**Discussion by irevoire** [meilisearch/product#725] [Experimental features]
Meilisearch v1.7.0 (released on March 11, 2024) introduces a new CLI parameter that changes the behavior of Meilisearch in a few ways that should help you run meilisearch in a cluster by externalizing the task queue.

## Enable the CLI flags 

```
./meilisearch --experimental-replication-parameters
```

## What this CLI flag does

This CLI flag changes three big things in the engine around the task queue:

### Disable the auto-deletion of tasks

By default, Meilisearch will auto-delete your latest tasks finished (succeeded, failed, or canceled) when you reach 1M tasks in your task queue.
When using this flag, you will be in charge of auto-deleting the tasks. Meilisearch won’t do it anymore.
But you **must** do it because once the task queue reaches 10GiB, Meilisearch will stop everyone from registering tasks.

### Chose your own task ID

When the `--experimental-replication-parameters` flag is enabled, you can specify your own task ID with the `TaskId` header instead of letting Meilisearch choose a new one, but there are a few rules around that:
- You must send the task ID in your request via the custom `TaskId` header (i.e., `TaskId: 0`)
- Your task ID must be a **valid number** 
- It must be **superior to the maximum task ID already seen in the task queue**. i.e., if you first register the `TaskId: 1000`, then you won’t be able to register the task id `1` or even `1000`; your task ID must be strictly `> 1000`.
- The task IDs are unsigned 32-bit numbers

### Dry-register a task

When the `--experimental-replication-parameters` flag is enabled, you also gain the ability to « dry register » a task.
That means, when you try to register a task, if you specify the header `DryRun: true` in your request, Meilisearch won’t register your task but will answer with a 200 and a validated task.

**Comment by macraig** [comment]
**Update:** [v1.7](https://github.com/meilisearch/meilisearch/releases/tag/v1.7.0) has been released 🐇

**Comment by curquiza** [comment]
Removed next Monday in v1.51.0
